G/GDAL/bundled/patches/bsd-environ-undefined-fix.patch

```diff
@@ -0,0 +1,27 @@
+diff --git a/gdal.cmake b/gdal.cmake
+index e221f99b79..bfd00348f3 100644
+--- a/gdal.cmake
++++ b/gdal.cmake
+@@ -277,14 +277,14 @@ endif ()
+ 
+ # Check that all symbols we need are present in our dependencies This is in particular useful to check that drivers
+ # built as plugins can access all symbols they need.
+-if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+-  include(CheckLinkerFlag)
+-  check_linker_flag(C "-Wl,--no-undefined" HAS_NO_UNDEFINED)
+-  if (HAS_NO_UNDEFINED AND (NOT "${CMAKE_CXX_FLAGS}" MATCHES "-fsanitize") AND NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
```

Comment:
Isn't this a matter of adding FreeBSD to this line instead of commenting out everything just for a single platform?

Reply:
I only apply the patch for freebsd.

Reply:
I know and that's precisely the problem, patches applied for a single platform are dodgy exactly because you can't apply them everywhere.  What I'm saying is that maybe this should have `AND NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD"` and can be safely applied everywhere.